src/js/controllers/imageCtrl-test.js

```diff
@@ -0,0 +1,133 @@
+                    isfamily: 0,
+                    isfriend: 0,
+                    ispublic: 1,
+                    owner: "138698049@N03",
+                    secret: "1d4dc255d0",
+                    server: "693",
+                    title: ""
+                }
+            ]
+        }
+
+    };
+    mockImageArray = ['https://farm6.staticflickr.com/5764/23780914905_0eb2a77a55_z.jpg', 'https://farm1.staticflickr.com/693/23698470321_1d4dc255d0_z.jpg'];
+
+    beforeEach(module('flickrPOC'));
+
+    describe('imagesCtrl', function(){
+
+        describe('Sucessful call to api', function() {
+            beforeEach(module(function ($provide) {
+                $provide.service('images', function ($q) {
+                    this.loadPics = sinon.stub().returns($q.resolve(sampleJSON));
+                    this.createImagesArray = sinon.stub().returns($q.resolve(mockImageArray));
+                });
+
+            }));
+
+            beforeEach(inject(function (_$rootScope_, _$controller_, _images_, _$q_, _$log_) {
+                $rootScope = _$rootScope_.$new();
+                $controller = _$controller_;
+                images = _images_;
+                $q = _$q_;
+                $log = _$log_;
+                vm = $controller('imagesCtrl');
+
+                $rootScope.$apply();
+
+            }));
+
+            describe('return flickr images from object from http call', function(){
+                var imageArray;
+
+                beforeEach(function(){
+                    imageArray = vm.images.$$state.value;
+                });
+
+               it('should return an array of images', function(){
+                   expect(angular.isArray(imageArray)).to.equal(true);
+               });
+
+               it('Should have all strings', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(angular.isString(value)).to.equal(true);
+                   });
+               });
+
+               it('Each strings must have an http address', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(value.substr(0, 8)).to.equal('https://');
+                   });
+               });
+
+               it('Each strings must have an flickr address', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(value).to.contain('.staticflickr.com');
+                   });
+               });
+
+               it('Each strings must have an flickr address', function(){
+                   angular.forEach(imageArray, function(value){
+                       expect(value.substr(-4, 4)).to.match(/^\.|\.jpg$|\.gif$|.png$/);
+                   });
+               });
+
+            });
+
+        });
+
+        describe('Error from to api', function() {
+            var errorObject = { "stat": "fail", "code": 2, "message": "Unknown user" }
+            var errorMsg = 'Unknown user';
```

Comment:
The `errorObject` is much better. Nice work. And I see what you're doing with the additional `errorMsg`. I have one suggestion though.

When you're working with mock data, you can *almost* always assume that someone will have to add to the data at some point in the future to handle specific use cases. To safe guide against this, I tend to use the mock object properties when asserting a value against the mock data.

For example, you have a test below: 
```javascript
it('Should log error message ', function(){
  expect($log.error.logs[0][0]).to.include(errorMsg);
});
```
This test is a valuable one to ensure that the $log service is called with the provided error message. No argument there. But what if someone changes the `errorObject.message` string, but overlooks the second reference to the message string (`errorMsg`)? An additional change would need to be made to match, duplicating the string. Not the best practice.

Instead, consider using the `errorObject` itself in your assertion, ensuring that any changes made will retain the passing test.
```javascript
it('Should log error message ', function(){
  expect($log.error.logs[0][0]).to.include(errorObject.message);
});
```